File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit = 0;


        int l = 0;


        for (int r = 1; r < prices.size(); r++) {
            if (prices[l] < prices[r]) {
                int profit = prices[r] - prices[l];
                maxProfit = std::max(maxProfit, profit);
            }
            else {
                l = r;
            }

        }
        return maxProfit;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l = 0
        r = len(heights) - 1
        curHeight = 0
        maxHeight = 0

        while l < r:
            curHeight = min(heights[l], heights[r]) * (r - l)
            maxHeight = max(curHeight, maxHeight)
            if heights[l] > heights[r]:
                # right pointer moves left
                r -= 1
            else:
                # left pointer moves right
                l += 1
        return maxHeight
